### PR TITLE
Add query param feature flags, 'Move to complete' button on course cards

### PR DESCRIPTION
**Pull request by adamstankiewicz**
This PR is part 1 of supporting the "Move to complete" UI functionality. It adds support for enabling features by specifying a query parameter in the URL, and displays the "Move to complete" button in the dropdown is the "move_to_complete" feature is enabled in the URL query parameter.

### CODE REVIEW

File: src/components/common/course-enrollments/course-cards/InProgressCourseCard.jsx [hunk 1]
```diff
@@ -26,8 +33,32 @@ const InProgressCourseCard = (props) => {
     return false;
   });
 
+  const getDropdownMenuItems = () => {
+    const isMoveToCompleteEnabled = isFeatureEnabled('move_to_completed');
+    const { pageContext: { pageType } } = useContext(LayoutContext);
+    if (!isMoveToCompleteEnabled || pageType !== 'pages.EnterprisePage') {
```
Comment (douglashall) [inline src/components/common/course-enrollments/course-cards/InProgressCourseCard.jsx]: Checking for `pageType` may be a sign that we need to change the way we are reusing code here. If would be a shame to litter the code with logic checks like this.

Comment (adamstankiewicz) [reply src/components/common/course-enrollments/course-cards/InProgressCourseCard.jsx]: Yeah, I was thinking the same thing. Maybe we can chat a bit about an approach that avoids these logic checks going forward?

**Comment by adamstankiewicz** [comment]
> Do we really need the `isFeatureEnabled` thing? Also, are we good with no tests? (I mean I am, for sure, just want to make sure you are as well).

In this case, the `isFeatureEnabled` thing probably isn't necessary, but only because we don't currently have enterprise users hitting this page yet. Assuming we did have users, it would be necessary in order to ship/test smaller chunks of work on the same feature before exposing it to users. Because it's something we'll likely want down the road (or some form of it), I opted to include it here as if we already had enterprise users hitting this page.

**Comment by douglashall** [comment]
> > Do we really need the `isFeatureEnabled` thing? Also, are we good with no tests? (I mean I am, for sure, just want to make sure you are as well).
> 
> In this case, the `isFeatureEnabled` thing probably isn't necessary, but only because we don't currently have enterprise users hitting this page yet. Assuming we did have users, it would be necessary in order to ship/test smaller chunks of work on the same feature before exposing it to users. Because it's something we'll likely want down the road (or some form of it), I opted to include it here as if we already had enterprise users hitting this page.

Makes sense. Maybe just wait until we need it then. That way you won't have to go back in and remove the `isFeatureEnabled` check later when we do go live.

**Comment by adamstankiewicz** [comment]
> Makes sense. Maybe just wait until we need it then. That way you won't have to go back in and remove the `isFeatureEnabled` check later when we do go live.

Fair enough 😃 I would still opt for at least leaving the functions in `common/features.js`. But then, remove the `isFeatureEnabled` check. That way, those functions exist when we do need them.

File: src/components/common/course-enrollments/course-cards/InProgressCourseCard.jsx (hunk 1 above)
Comment (douglashall) [reply src/components/common/course-enrollments/course-cards/InProgressCourseCard.jsx]: One way would be to pass the dropdown menu items as props down the component hierarchy from the enterprise-specific parent component.

**Pull request merged**